scout/server/blueprints/cases/views.py

```diff
@@ -74,9 +74,32 @@ def index():
 
 
 @cases_bp.route("/<institute_id>/<case_name>")
+@cases_bp.route("/case/case_id/<case_id>")
 @templated("cases/case.html")
-def case(institute_id, case_name):
-    """Display one case."""
+def case(
+    case_name: Optional[str] = None,
+    institute_id: Optional[str] = None,
+    case_id: Optional[str] = None,
+):
+    """Display one case.
+    Institute and display_name pairs uniquely specify a case.
+    So do case_id, but we still call institute_and_case again to fetch institute
+    and reuse its user access verification.
+    """
+
+    if not institute_id:
```

Comment:
Why not just:

```
if case_id:
   case_obj = store.case(case_id=case_id, projection={"display_name": 1, "owner": 1})
   case_name = case_obj.get("display_name")
   institute_id = case_obj.get("owner")
```
Then you just go to line 103?

Reply:
Which is the same except for removing the error handling? 😉

Reply:
It's shorter, and if the case doesn't exist or you have no access to it already handled on line 106

Reply:
Removing the double indentation ifs is good for complexity, agreed, also for the price of loosing one error check I think. But having to have multiple if - else on the right side (those post hocs are slow for me to read at least), and loosing another does not seem to justify removing the other `if not`.

If we want it really simple, I could also just add another endpoint for it. That would cause a few lines of duplication, but keep each very easy to read.